gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/writer/FileAwareInputStreamDataWriter.java

```diff
@@ -500,15 +504,26 @@ private void ensureDirectoryExists(FileSystem fs, Path path, Iterator<OwnerAndPe
 
       String group = ownerAndPermission.getGroup();
       String owner = ownerAndPermission.getOwner();
+      List<AclEntry> aclEntries = ownerAndPermission.getAclEntries();
       if (group != null || owner != null) {
         log.debug("Applying owner {} and group {} to path {}.", owner, group, path);
         fs.setOwner(path, owner, group);
       }
+      if (!aclEntries.isEmpty()) {
+        setAclOnPath(fs, path, aclEntries);
+      }
     } else {
       fs.mkdirs(path);
     }
   }
 
+  /*
+   * Creating an abstraction layer to support unit testing on raw local file system for validating if ACLs are set on a file path
```

Comment:
@Will-Lo asked below, which probably indicates this is not stated clearly enough.  the motivation arises because the `LocalFileSystem` lacks support--do be sure to say so

also, this should be javadoc comment to show up there.

Reply:
I no longer use this stubbing and have extended the LocalFileSystem to support the setAcl method.